blocks/library/image/index.js

```diff
@@ -149,17 +153,22 @@ registerBlockType( 'core/image', {
 	},
 
 	save( { attributes } ) {
-		const { url, alt, caption, align = 'none' } = attributes;
+		const { url, alt, caption, align = 'none', href } = attributes;
 		const needsWrapper = [ 'wide', 'full' ].indexOf( align ) !== -1;
 
 		// If there's no caption set only save the image element.
 		if ( ! needsWrapper && ( ! caption || ! caption.length ) ) {
-			return <img src={ url } alt={ alt } className={ `align${ align }` } />;
+			return href
+				? <a href={ href }><img src={ url } alt={ alt } className={ `align${ align }` } /></a>
+				: <img src={ url } alt={ alt } className={ `align${ align }` } />;
```

Comment:
Ideally we could avoid some duplication here by assigning the base `img` tag to a variable, and wrap it only if the `href` condition applies.